mlflow/tracking/fluent.py

```diff
@@ -303,7 +304,6 @@ def _get_experiment_id_from_env():
 
 
 def _get_experiment_id():
-    return int(_active_experiment_id or
-               _get_experiment_id_from_env() or
-               (is_in_databricks_notebook() and get_notebook_id()) or
-               Experiment.DEFAULT_EXPERIMENT_ID)
+    return (_active_experiment_id or
+            _get_experiment_id_from_env() or
+            (is_in_databricks_notebook() and get_notebook_id())) or None
```

Comment:
We may want to release a version, 0.9.1, where we continue to pass default experiment id. This would allow the new client to talk to old servers, and then in 1.0 we can change the client behavior to use a None experiment id.